atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java

```diff
@@ -2097,35 +2114,36 @@ private Transaction getSnapshotTransactionWith(
-                        metricsManager,
-                        keyValueServiceWrapper.apply(keyValueService, pathTypeTracker),
-                        timelockService,
-                        NoOpLockWatchManager.create(),
-                        transactionService,
-                        NoOpCleaner.INSTANCE,
-                        startTs,
-                        TestConflictDetectionManagers.createWithStaticConflictDetection(
-                                ImmutableMap.of(TABLE, Optional.of(ConflictHandler.RETRY_ON_WRITE_WRITE))),
-                        SweepStrategyManagers.createDefault(keyValueService),
-                        lockImmutableTimestampResponse.getImmutableTimestamp(),
-                        Optional.of(lockImmutableTimestampResponse.getLock()),
-                        preCommitCondition,
-                        AtlasDbConstraintCheckingMode.NO_CONSTRAINT_CHECKING,
-                        null,
-                        TransactionReadSentinelBehavior.THROW_EXCEPTION,
-                        false,
-                        timestampCache,
-                        getRangesExecutor,
-                        defaultGetRangesConcurrency,
-                        MultiTableSweepQueueWriter.NO_OP,
-                        MoreExecutors.newDirectExecutorService(),
-                        validateLocksOnReads,
-                        () -> transactionConfig,
-                        ConflictTracer.NO_OP,
-                        tableLevelMetricsController),
-                pathTypeTracker);
+        SnapshotTransaction transaction = new SnapshotTransaction(
+                metricsManager,
+                keyValueServiceWrapper.apply(keyValueService, pathTypeTracker),
+                timelockService,
+                timelockServices.getLockWatchManager(),
+                transactionService,
+                NoOpCleaner.INSTANCE,
+                startTs,
+                TestConflictDetectionManagers.createWithStaticConflictDetection(
+                        ImmutableMap.of(TABLE, Optional.of(ConflictHandler.RETRY_ON_WRITE_WRITE))),
+                SweepStrategyManagers.createDefault(keyValueService),
+                lockImmutableTimestampResponse.getImmutableTimestamp(),
+                Optional.of(lockImmutableTimestampResponse.getLock()),
+                preCommitCondition,
+                AtlasDbConstraintCheckingMode.NO_CONSTRAINT_CHECKING,
+                null,
+                TransactionReadSentinelBehavior.THROW_EXCEPTION,
+                false,
+                timestampCache,
+                getRangesExecutor,
+                defaultGetRangesConcurrency,
+                MultiTableSweepQueueWriter.NO_OP,
+                MoreExecutors.newDirectExecutorService(),
+                validateLocksOnReads,
+                () -> transactionConfig,
+                ConflictTracer.NO_OP,
+                tableLevelMetricsController);
+        // hack (?), tracking lock
```

Comment:
can you elaborate?

Reply:
leftover from test-debugging pain; removed